server/fidelity/macros/set_german_and_publish.py

```diff
@@ -99,6 +120,119 @@ def set_german_and_publish(item, **kwargs):
+                    de_group_root = next(g for g in de_item_groups if g['id'] == 'root')
+                    de_group_root['refs'].append({"idRef": ori_group_id})
+                    en_de_map = {}
+
+                # we synchronise new items
+                new_refs = ori_refs - en_de_map.keys()
+                for new_ref in new_refs:
+                    # we check if we have already a duplicate version of this item
+                    dup_ref_item = archive_service.find_one(req=None, original_id=new_ref, language="de")
+                    if dup_ref_item is None:
+                        # we have to duplicate the item, basically reproducing what would happen
+                        # with publish workflow
+                        logger.info("creating German item for {item_id}".format(item_id=new_ref))
+                        ori_ref_item = archive_service.find_one(req=None, guid=new_ref)
+                        # we imitate internal destination
+                        ori_ref_item_cpy = deepcopy(ori_ref_item)
+                        send_to(ori_ref_item_cpy, desk_id=dest_desk_id, stage_id=dest_stage_id)
+                        new_id, updates = create_de_item(archive_service, ori_ref_item_cpy)
+                        dup_ref_item = archive_service.find_one(req=None, guid=new_id)
+                        updates[LINKED_IN_PACKAGES] = dup_ref_item.setdefault(LINKED_IN_PACKAGES, [])
+                        dup_ref_item[PUBLISHED_IN_PACKAGE] = updates[PUBLISHED_IN_PACKAGE] = de_item_id
+                        updates[LINKED_IN_PACKAGES].append({PACKAGE: de_item_id})
+
+                        archive_publish_service.patch(id=new_id, updates=updates)
+                        insert_into_versions(id_=new_id)
+                    else:
+                        # the item already exists, let's update it
+                        updates = {
+                            LINKED_IN_PACKAGES: dup_ref_item.setdefault(LINKED_IN_PACKAGES, []),
+                            PUBLISHED_IN_PACKAGE: de_item_id,
+                        }
+                        updates[LINKED_IN_PACKAGES].append({PACKAGE: de_item_id})
+                        archive_service.system_update(id=dup_ref_item["guid"], updates=updates, original=dup_ref_item)
+
+                    # we have the German item, now we create the ref to update German package
+                    new_ref = get_item_ref(dup_ref_item)
+                    new_ref['guid'] = new_ref[RESIDREF]
+                    de_group['refs'].append(new_ref)
+
+                # now we remove items removed from English version
+                deleted_refs = en_de_map.keys() - ori_refs
+                for deleted_ref in deleted_refs:
+                    de_deleted_ref = en_de_map[deleted_ref]
+                    for ref in de_group['refs']:
+                        if ref.get('guid') == de_deleted_ref:
+                            break
+                    de_group['refs'].remove(ref)
```

Comment:
shouldn't it be
```
for ref in de_group['refs']:
    if ref.get('guid') == de_deleted_ref:
        break
    de_group['refs'].remove(ref)
```
?

Reply:
no this is fine, it must be removed when it's matching the `de_delete_ref`. To make it more explicit, I've moved it above the `break` now (will push soon).